src/main/scala/scorex/core/api/http/PeersApiRoute.scala

```diff
@@ -41,10 +40,13 @@ case class PeersApiRoute(peerManager: ActorRef,
       _.flatMap { con =>
         con.peerInfo.map { peerInfo =>
           PeerInfoResponse(
-            address = peerInfo.peerSpec.declaredAddress.map(_.toString).getOrElse(""),
+            remoteAddress = con.connectionId.remoteAddress.toString,
+            localAddress = Some(con.connectionId.localAddress.toString),
```

Comment:
What is the difference between `peerSpec.declaredAddress` and `con.connectionId.remoteAddress/localAddress` ?
Is declared one always equal to remote or local?

Note, when this change will be integrated into the SDK, we need to check the STF tests, I think, we rely on `address` field when connect the nodes

Reply:
- `peerSpec.declaredAddress` represents the declared address in the config file, it can be `None`;
- `con.connectionId.remoteAddress/localAddress` are values passed by the `TcpManager` akka actor.
If the declaredAddress is correctly set, then `peerSpec.declaredAddress == con.connectionId.remoteAddress`; remote and local address could be the same; in our case is not, since the remote is the machine's address the node is running on, the local address will be the internal docker network address.

About the last note, with the STF tests you mean the python tests in the SDK?

Reply:
I mean usage of this utility method: https://github.com/HorizenOfficial/Sidechains-SDK/pull/449/files#diff-4691d17d8020a16aae411a61ec7aa9ec7f56a033a9152ae78d122ce4f3c9cd2aR603

Reply:
yes, that address will need to be renamed to remoteAddress. This opens an interesting question: do we want to start supporting API endpoints versioning? I'm unaware of any in the SDK, but maybe I'm wrong and it's already supported. Perhaps this endpoint is not called by anyone but the python tests so we can change the response, but I think it's something worth taking into account

Reply:
at some point - we will, for now - just notifying about the changes.